src/searchModal.js

```diff
@@ -158,6 +161,26 @@ export default function searchModalFactory(config) {
         });
     }
 
+    /**
+     * Loops through each class in received class tree and sets access mode to 'denied' on private classes
+     * so are disabled on class tree
+     * @param {object} classTree - class tree received by server, containing resources (classes) and permissions
+     */
+    function manageClassTreePermissions(classTree) {
+        const disableBlockedClasses = function (resources) {
+            resources.forEach((resource, index, array) => {
+                if (classTree.permissions.data[resource.uri].find(permission => permission === 'READ')) {
```

Comment:
**though:** Are you sure the data is there? Won't it throw a runtime issue?

Reply:
It seems to always be there for this case of use, but after checking [other implementations](https://github.com/oat-sa/tao-core/blob/c2e5436443deee674241e77c9530f184105e1328/views/js/provider/resources.js#L151), now I check it first.